bin/citrea/tests/e2e/proving.rs

```diff
@@ -187,3 +187,178 @@ async fn full_node_verify_proof_and_store() {
+            GenesisPaths::from_dir(TEST_DATA_GENESIS_PATH),
+            Some(ProverConfig {
+                proving_mode: sov_stf_runner::ProverGuestRunConfig::Execute,
+                // Make it impossible for proving to happen
+                proof_sampling_number: 1_000_000,
+                enable_recovery: true,
+            }),
+            rollup_config,
+            None,
+        )
+        .await;
+    });
+
+    let prover_node_port = prover_node_port_rx.await.unwrap();
+
+    let prover_node_test_client = make_test_client(prover_node_port).await.unwrap();
+
+    let (full_node_port_tx, full_node_port_rx) = tokio::sync::oneshot::channel();
+
+    let rollup_config = create_default_rollup_config(
+        true,
+        &fullnode_db_dir,
+        &da_db_dir,
+        NodeMode::FullNode(seq_port),
+    );
+    let full_node_task = tokio::spawn(async {
+        start_rollup(
+            full_node_port_tx,
+            GenesisPaths::from_dir(TEST_DATA_GENESIS_PATH),
+            None,
+            rollup_config,
+            None,
+        )
+        .await;
+    });
+
+    let full_node_port = full_node_port_rx.await.unwrap();
+    let full_node_test_client = make_test_client(full_node_port).await.unwrap();
+
+    da_service.publish_test_block().await.unwrap();
+    wait_for_l1_block(&da_service, 2, None).await;
+
+    test_client.send_publish_batch_request().await;
+    test_client.send_publish_batch_request().await;
+    test_client.send_publish_batch_request().await;
+    test_client.send_publish_batch_request().await;
+    wait_for_l2_block(&full_node_test_client, 4, None).await;
+
+    // Commitment submitted
+    wait_for_l1_block(&da_service, 3, None).await;
+
+    // Full node sync commitment block
+    test_client.send_publish_batch_request().await;
+    wait_for_l2_block(&full_node_test_client, 5, None).await;
+
+    // Trigger proving via the RPC endpoint
+    prover_node_test_client.prover_prove(3, Some(true)).await;
+
+    // wait here until we see from prover's rpc that it finished proving
+    wait_for_prover_l1_height(&prover_node_test_client, 4, None)
+        .await
+        .unwrap();
+
+    let commitments = prover_node_test_client
+        .ledger_get_sequencer_commitments_on_slot_by_number(3)
+        .await
+        .unwrap()
+        .unwrap();
+    assert_eq!(commitments.len(), 1);
+
+    assert_eq!(commitments[0].l2_start_block_number, 1);
+    assert_eq!(commitments[0].l2_end_block_number, 4);
+
+    assert_eq!(commitments[0].found_in_l1, 3);
+
+    let third_block_hash = da_service.get_block_at(3).await.unwrap().header.hash;
+
+    let commitments_hash = prover_node_test_client
+        .ledger_get_sequencer_commitments_on_slot_by_hash(third_block_hash.0)
+        .await
+        .unwrap()
+        .unwrap();
+    assert_eq!(commitments_hash, commitments);
+
+    let prover_proof = prover_node_test_client
+        .ledger_get_proofs_by_slot_height(3)
+        .await[0]
+        .clone();
```

Comment:
from this I understand that we record proofs and related info to ledger db, right? we should do it so good.

Reply:
Yea that was what the test in the same file is doing so i did the same.